### PR TITLE
DlgAddkey

### PR DESCRIPTION
**Pull request by maxwellfundi**
@africanmathsinitiative/developers  this is ready for review

### CODE REVIEW

File: instat/dlgAddKey.vb [hunk 1]
```diff
@@ -18,42 +18,63 @@ Imports instat.Translations
...
-        clsAnyDuplicatesFunction.SetRCommand("anyDuplicated")
-        clsSumFunction.SetRCommand("sum")
-        clsIsNAFunction.SetRCommand("is.na")
-        clsSumFunction.AddParameter("x", clsRFunctionParameter:=clsIsNAFunction)
     End Sub
 
     Private Sub SetDefaults()
         ucrSelectorKeyColumns.Reset()
         ucrInputKeyName.ResetText()
         ucrInputCheckInput.ResetText()
```
Comment (lilyclements) [inline instat/dlgAddKey.vb]: Is this a rare case we are allowed to use ResetText? Or should this be something else?

File: instat/dlgAddKey.vb [hunk 2]
```diff
@@ -125,7 +134,12 @@ Public Class dlgAddKey
         TestOKEnabled()
     End Sub
 
-    Private Sub ucrInputKeyName_ContentsChanged() Handles ucrInputKeyName.ContentsChanged
+    Private Sub AllControls_ControlContentsChanged() Handles ucrInputKeyName.ControlContentsChanged, ucrReceiverKeyColumns.ControlContentsChanged
+        TestOKEnabled()
+    End Sub
+
+    Private Sub ucrBase_ClickReset(sender As Object, e As EventArgs) Handles ucrBase.ClickReset
+        SetDefaults()
```
Comment (lilyclements) [inline instat/dlgAddKey.vb]: You forgot your SetRCodeforControls

File: instat/dlgAddKey.vb (hunk 1 above)
Comment (maxwellfundi) [inline instat/dlgAddKey.vb]: Hahaha this should be reset actually.

File: instat/dlgAddKey.vb (hunk 2 above)
Comment (maxwellfundi) [inline instat/dlgAddKey.vb]: I think i keep forgetting to include this here.

Fixing those shortly. 

Thanks.

File: instat/dlgAddKey.vb (hunk 1 above)
Comment (lilyclements) [inline instat/dlgAddKey.vb]: But, I believe just doing reset won't clear it? Or does it?

Comment (maxwellfundi) [inline instat/dlgAddKey.vb]: We use reset for ucr Inputs

**Comment by maxwellfundi** [comment]
@lilyclements  have a look at this now. 

The dlgAddlink was accidentally edited there and didnt notice that actually. 

I see the it has a small conflict which @dannyparsons could fix quite easily.

**Comment by maxwellfundi** [comment]
Closing this to open a new pull request on the same

**Pull request closed without merging**